src/request.js

```diff
@@ -63,7 +63,7 @@ function replyImmediately(request, response) {
 function handleMessageWrapper(message, request) {
 	log(5, "Core received message");
 
-	let response = JSON.parse(message);
+	let response = typeof message === 'string' ? JSON.parse(message) : message;
```

Comment:
I'm thinking this is a decent assumption to make, but let me know if this might break something. All the unit tests pass